<?xml version="1.0"?>
<robot name="robot_name">
<!-- define link -->
    <link name="base_link">
        <!-- <inertial>
            <origin xyz="0.0 0.0 0.05" rpy="0.0 0.0 0.0"/>
            <mass value="1.0"/>
            <inertia ixx="0.01" ixy="0.0" ixz="0.0" iyy="0.01" iyz="0.0" izz="0.001"/>
        </inertial> -->
        <visual>
            <geometry>
                <box size="0.3 0.3 0.1"/>
            </geometry>
            <origin xyz="0.0 0.0 0.05" rpy="0.0 0.0 0.0"/>
            <material name="blue">
                <color rgba="0.0 0.0 1.0 1.0"/>
            </material>
        </visual>
        <collision>
            <geometry>
                <box size="0.3 0.3 0.1"/>
            </geometry>
            <origin xyz="0.0 0.0 0.05" rpy="0.0 0.0 0.0"/>
        </collision>
    </link>
     <link name="link1">
        <inertial>
            <origin xyz="0.0 0.0 0.05" rpy="0.0 0.0 0.0"/>
            <mass value="1.0"/>
            <inertia ixx="0.01" ixy="0.0" ixz="0.0" iyy="0.01" iyz="0.0" izz="0.001"/>
        </inertial>
        <visual>
            <geometry>
                <cylinder radius="0.05" length="0.1"/>
            </geometry>
            <origin xyz="0.0 0.0 0.05" rpy="0.0 0.0 0.0"/>
            <material name="pink">
                <color rgba="1.0 0.0 1.0 1.0"/>
            </material>
        </visual>
        <collision>
            <geometry>
                <cylinder radius="0.05" length="0.1"/>
            </geometry>
            <origin xyz="0.0 0.0 0.05" rpy="0.0 0.0 0.0"/>
        </collision>
    </link>
    <link name="link2">
        <inertial>
            <origin xyz="0.0 0.0 0.15" rpy="0.0 0.0 0.0"/>
            <mass value="1.0"/>
            <inertia ixx="0.01" ixy="0.0" ixz="0.0" iyy="0.01" iyz="0.0" izz="0.001"/>
        </inertial>
        <visual>
            <geometry>
                <cylinder radius="0.05" length="0.3"/>
            </geometry>
            <origin xyz="0.0 0.0 0.15" rpy="0.0 0.0 0.0"/>
            <material name="red">
                <color rgba="1.0 0.0 0.0 1.0"/>
            </material>
        </visual>
        <collision>
            <geometry>
                <cylinder radius="0.05" length="0.3"/>
            </geometry>
            <origin xyz="0.0 0.0 0.15" rpy="0.0 0.0 0.0"/>
        </collision>
    </link>
    <link name="link3">
        <inertial>
            <origin xyz="0.0 0.0 0.15" rpy="0.0 0.0 0.0"/>
            <mass value="1.0"/>
            <inertia ixx="0.01" ixy="0.0" ixz="0.0" iyy="0.01" iyz="0.0" izz="0.001"/>
        </inertial>
        <visual>
            <geometry>
                <cylinder radius="0.05" length="0.3"/>
            </geometry>
            <origin xyz="0.0 0.0 0.15" rpy="0.0 0.0 0.0"/>
            <material name="yellow">
                <color rgba="1.0 1.0 0.0 1.0"/>
            </material>
        </visual>
        <collision>
            <geometry>
                <cylinder radius="0.05" length="0.3"/>
            </geometry>
            <origin xyz="0.0 0.0 0.15" rpy="0.0 0.0 0.0"/>
        </collision>
    </link>
    <link name="link4">
        <inertial>
            <origin xyz="0.0 0.0 0.15" rpy="0.0 0.0 0.0"/>
            <mass value="1.0"/>
            <inertia ixx="0.01" ixy="0.0" ixz="0.0" iyy="0.01" iyz="0.0" izz="0.001"/>
        </inertial>
        <visual>
            <geometry>
                <cylinder radius="0.05" length="0.3"/>
            </geometry>
            <origin xyz="0.0 0.0 0.15" rpy="0.0 0.0 0.0"/>
            <material name="green">
                <color rgba="0.0 1.0 0.0 1.0"/>
            </material>
        </visual>
        <collision>
            <geometry>
                <cylinder radius="0.05" length="0.3"/>
            </geometry>
            <origin xyz="0.0 0.0 0.15" rpy="0.0 0.0 0.0"/>
        </collision>
     </link>
    <link name="link5">
        <inertial>
            <origin xyz="0.0 0.0 0.15" rpy="0.0 0.0 0.0"/>
            <mass value="1.0"/>
            <inertia ixx="0.01" ixy="0.0" ixz="0.0" iyy="0.01" iyz="0.0" izz="0.001"/>
        </inertial>
        <visual>
            <geometry>
                <cylinder radius="0.05" length="0.3"/>
            </geometry>
            <origin xyz="0.0 0.0 0.15" rpy="0.0 0.0 0.0"/>
            <material name="red">
                <color rgba="1.0 0.0 0.0 1.0"/>
            </material>
        </visual>
        <collision>
            <geometry>
                <cylinder radius="0.05" length="0.3"/>
            </geometry>
            <origin xyz="0.0 0.0 0.15" rpy="0.0 0.0 0.0"/>
        </collision>
     </link>
    <link name="gripper_base_link">
        <inertial>
            <origin xyz="0.0 0.0 0.05" rpy="0.0 0.0 0.0"/>
            <mass value="1.0"/>
            <inertia ixx="0.01" ixy="0.0" ixz="0.0" iyy="0.01" iyz="0.0" izz="0.001"/>
        </inertial>
        <visual>
            <geometry>
                <box size="0.3 0.1 0.1"/>
            </geometry>
            <origin xyz="0.0 0.0 0.05" rpy="0.0 0.0 0.0"/>
            <material name="black">
                <color rgba="0.0 0.0 0.0 1.0"/>
            </material>
        </visual>
        <collision>
            <geometry>
                <box size="0.3 0.1 0.1"/>
            </geometry>
            <origin xyz="0.0 0.0 0.05" rpy="0.0 0.0 0.0"/>
        </collision>
     </link>
    <link name="hand_left_link">
        <inertial>
            <origin xyz="0.0 0.0 0.05" rpy="0.0 0.0 0.0"/>
            <mass value="1.0"/>
            <inertia ixx="0.01" ixy="0.0" ixz="0.0" iyy="0.01" iyz="0.0" izz="0.001"/>
        </inertial>
        <visual>
            <geometry>
                <box size="0.05 0.1 0.1"/>
            </geometry>
            <origin xyz="0.0 0.0 0.05" rpy="0.0 0.0 0.0"/>
            <material name="pink">
                <color rgba="1.0 0.0 1.0 1.0"/>
            </material>
        </visual>
        <collision>
            <geometry>
                <box size="0.05 0.1 0.1"/>
            </geometry>
            <origin xyz="0.0 0.0 0.05" rpy="0.0 0.0 0.0"/>
        </collision>
     </link>
    <link name="hand_right_link">
        <inertial>
            <origin xyz="0.0 0.0 0.05" rpy="0.0 0.0 0.0"/>
            <mass value="1.0"/>
            <inertia ixx="0.01" ixy="0.0" ixz="0.0" iyy="0.01" iyz="0.0" izz="0.001"/>
        </inertial>
        <visual>
            <geometry>
                <box size="0.05 0.1 0.1"/>
            </geometry>
            <origin xyz="0.0 0.0 0.05" rpy="0.0 0.0 0.0"/>
            <material name="pink">
                <color rgba="1.0 0.0 1.0 1.0"/>
            </material>
        </visual>
        <collision>
            <geometry>
                <box size="0.05 0.1 0.1"/>
            </geometry>
            <origin xyz="0.0 0.0 0.05" rpy="0.0 0.0 0.0"/>
        </collision>
     </link>
<!-- define joint -->
    <joint name="joint1" type="revolute">
        <parent link="base_link"/>
        <child link="link1"/>
        <origin xyz="0.0 0.0 0.1" rpy="0.0 0.0 0.0"/>
        <axis xyz="0.0 0.0 1.0"/>
        <limit lower="-3.14" upper="3.14" effort="100.0" velocity="1.0"/>
    </joint>
    <joint name="joint2" type="revolute">
        <parent link="link1"/>
        <child link="link2"/>
        <origin xyz="0.0 0.0 0.1" rpy="0.0 0.0 0.0"/>
        <axis xyz="1.0 0.0 0.0"/>
        <limit lower="-3.14" upper="3.14" effort="100.0" velocity="1.0"/>
    </joint>
    <joint name="joint3" type="revolute">
        <parent link="link2"/>
        <child link="link3"/>
        <origin xyz="0.0 0.0 0.3" rpy="0.0 0.0 0.0"/>
        <axis xyz="1.0 0.0 0.0"/>
        <limit lower="-3.14" upper="3.14" effort="100.0" velocity="1.0"/>
    </joint>
    <joint name="joint4" type="revolute">
        <parent link="link3"/>
        <child link="link4"/>
        <origin xyz="0.0 0.0 0.3" rpy="0.0 0.0 0.0"/>
        <axis xyz="1.0 0.0 0.0"/>
        <limit lower="-3.14" upper="3.14" effort="100.0" velocity="1.0"/>
    </joint>
    <joint name="joint5" type="revolute">
        <parent link="link4"/>
        <child link="link5"/>
        <origin xyz="0.0 0.0 0.3" rpy="0.0 0.0 0.0"/>
        <axis xyz="1.0 0.0 0.0"/>
        <limit lower="-3.14" upper="3.14" effort="100.0" velocity="1.0"/>
    </joint>
    <joint name="joint6" type="revolute">
        <parent link="link5"/>
        <child link="gripper_base_link"/>
        <origin xyz="0.0 0.0 0.3" rpy="0.0 0.0 0.0"/>
        <axis xyz="0.0 0.0 1.0"/>
        <limit lower="-3.14" upper="3.14" effort="100.0" velocity="1.0"/>
    </joint>
    <joint name="joint_gripper_and_left" type="prismatic">
        <parent link="gripper_base_link"/>
        <child link="hand_left_link"/>
        <origin xyz="0.125 0.0 0.1" rpy="0.0 0.0 0.0"/>
        <axis xyz="-1.0 0.0 0.0"/>
        <limit lower="0.0" upper="0.1" effort="3000.0" velocity="0.1"/>
    </joint>
    <joint name="joint_gripper_and_right" type="prismatic">
        <parent link="gripper_base_link"/>
        <child link="hand_right_link"/>
        <origin xyz="-0.125 0.0 0.1" rpy="0.0 0.0 0.0"/>
        <axis xyz="1.0 0.0 0.0"/>
        <limit lower="0.0" upper="0.1" effort="100.0" velocity="0.1"/>
        <mimic joint="joint_gripper_and_left"></mimic>
    </joint>
</robot>